pkg/controller/utils/comparison/comparison.go

```diff
@@ -73,3 +74,26 @@ func StringsContains(a []string, x string) bool {
 	}
 	return false
 }
+
+// GenerateHashFromNodeAnnotation use to generate the MD5 hash from Node annotations
+func GenerateHashFromNodeAnnotation(edsNamespace, edsName string, nodeAnnotations map[string]string) string {
+	// build prefix for this specific eds
+	prefixKey := fmt.Sprintf(datadoghqv1alpha1.ExtendedDaemonSetRessourceNodeAnnotationKey, edsNamespace, edsName, "")
+
+	resourcesAnnotations := []string{}
+	for key, value := range nodeAnnotations {
+		if strings.HasPrefix(key, prefixKey) {
+			resourcesAnnotations = append(resourcesAnnotations, fmt.Sprintf("%s=%s", key, value))
+		}
+	}
```

Comment:
nit, we can hash in this block and avoid allocating `resourcesAnnotations`

Reply:
I didn't done it in this loop, because the order in `range map` is not always the same.

other implementation that I can do: store and sort only the keys in a slice, and loop a second time on the map? do yo think it will be a big win?

Reply:
okay but we're not sorting `resourcesAnnotations` before calculating the hash and we're populating the slice with the random order of the map. I think we need to sort the slice before hashing

Reply:
I remove the line by mistake :)
pr updated